.gitlab-ci.yml

```diff
@@ -359,6 +359,7 @@ Publish CP podspecs (dependent):
     - !reference [.release-pipeline-20m-delayed-job, rules]
```

Comment:
Do we still need the 20 min delay? This seems redundant and could potentially delay the pipeline unnecessarily.

Reply:
I had this same question!
It seems that after publishing to the CDN with `pod trunk push --synchronous *`, the new version of the pod is not immediately available. Some users have [mentioned](https://stackoverflow.com/questions/41702187/how-long-does-it-take-for-a-new-custom-cocoapod-to-get-uploaded-to-the-pod-trunk) that it may take a while before it becomes accessible via `pod install --repo-update` or `pod repo update`.

Reply:
Yes, it takes a while... The best source of truth are the [CP specs](https://github.com/CocoaPods/Specs/tree/master/Specs/d/2/d/DatadogCore) so far. As you can see, even the [website](https://cocoapods.org/pods/DatadogSDK) is not up to date; we released v2.20 3 weeks ago, and yet it shows v2.16.

Reply:
I remember @ncreated added these delays, so he's probably the one with more context on this.

Reply:
@simaoseica-dd is correct — after pod `trunk push --synchronous *` completes, it typically takes around 20 minutes for the pod to become available for use by any pod command, even with `--repo-update`. If we attempt to push dependent pods before this, pod trunk push will fail.

Currently, our release jobs are [designed to be **failure-resistant**](https://github.com/DataDog/dd-sdk-ios/blob/a4c6a1cb49a22eec9f2d5dada8be160082533d35/tools/release/publish-podspec.sh#L213-L218). If a pod isn't yet available and pod trunk push fails, the job won’t fail outright; instead, it will wait for a minute and then retry the push.

This means that while the 20-minute delay is not strictly required in our automation, it's a safeguard. Our CI jobs are constrained by a 1-hour timeout. If we don't introduce the delay, we risk running out of time, leaving only 40 minutes to push all podspecs — which may not be sufficient to complete the entire process.